include/drivers/pio.h

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright © 2017-2018 The Crust Firmware Authors.
+ * SPDX-License-Identifier: (BSD-3-Clause OR GPL-2.0)
+ */
+
+#ifndef DRIVERS_PIO_H
+#define DRIVERS_PIO_H
+
+#include <dm.h>
+#include <stdbool.h>
+#include <stdint.h>
+
+#define PIO_OPS(dev) ((struct pio_driver_ops *)((dev)->drv->ops))
+
+struct pio_driver_ops {
+	bool (*read_pin)(struct device *dev, uint8_t pin);
+	int  (*set_mode)(struct device *dev, uint8_t pin, uint8_t mode);
+	int  (*write_pin)(struct device *dev, uint8_t pin, uint8_t val);
+};
+
+/**
+ * Read the value of a pin in the Port L Data Register.
+ *
+ * @param dev The R_PIO device.
```

Comment:
"The port I/O device", and same below. (`R_PIO` is just one port controller.)

Reply:
I believe the new description is what we are going for. Please let me know what you think.